test/navigationLibrary/StandardRoute/StandardRouteWaypointModel.spec.js

```diff
@@ -69,48 +70,65 @@ ava('.toWaypointModel() returns a new instance of a WaypointModel', t => {
+    const modelWithAltitude = new StandardRouteWaypointModel(['BAKRR', 'A100+|A140-']);
+    const modelWithSpeed = new StandardRouteWaypointModel(['BAKRR', 'S210+|S250-']);
+    const modelWithAltitudeAndSpeed = new StandardRouteWaypointModel(['BAKRR', 'A100+|A140-|S210+|S250-']);
+
+    t.true(modelWithAltitude.altitudeMinimum === 10000);
+    t.true(modelWithAltitude.altitudeMaximum === 14000);
+    t.true(modelWithSpeed.speedMinimum === 210);
+    t.true(modelWithSpeed.speedMaximum === 250);
+    t.true(modelWithAltitudeAndSpeed.altitudeMinimum === 10000);
+    t.true(modelWithAltitudeAndSpeed.altitudeMaximum === 14000);
+    t.true(modelWithAltitudeAndSpeed.speedMinimum === 210);
+    t.true(modelWithAltitudeAndSpeed.speedMaximum === 250);
 });
 
-ava('._parseWaypointRestrictions() extracts an alititude restriction from a waypointRestrictions string by calling ._setAltitudeRestriction()', t => {
-    const model = new StandardRouteWaypointModel(['BAKRR', 'A80+']);
-    const spy = sinon.spy(model, '_setAltitudeRestriction');
-
-    model._parseWaypointRestrictions('A180+');
-
-    t.true(spy.callCount === 1);
-    t.true(model._altitude === 18000);
-    t.true(model._speed === -1);
+ava('._applyRestrictions() extracts all restrictions when non-ranged "AT" restrictions are used', (t) => {
+    const modelWithAltitudeRange = new StandardRouteWaypointModel(['BAKRR', 'A100']);
+    const modelWithSpeedRange = new StandardRouteWaypointModel(['BAKRR', 'S210']);
+    const modelWithAltitudeRangeAndSpeedRange = new StandardRouteWaypointModel(['BAKRR', 'A100|S210']);
+
+    t.true(modelWithAltitudeRange.altitudeMinimum === 10000);
+    t.true(modelWithAltitudeRange.altitudeMaximum === 10000);
+    t.true(modelWithSpeedRange.speedMinimum === 210);
+    t.true(modelWithSpeedRange.speedMaximum === 210);
+    t.true(modelWithAltitudeRangeAndSpeedRange.altitudeMinimum === 10000);
+    t.true(modelWithAltitudeRangeAndSpeedRange.altitudeMaximum === 10000);
+    t.true(modelWithAltitudeRangeAndSpeedRange.speedMinimum === 210);
+    t.true(modelWithAltitudeRangeAndSpeedRange.speedMaximum === 210);
 });
 
-ava('._parseWaypointRestrictions() extracts a speed restriction from a waypointRestrictions string by calling ._setSpeedRestriction()', t => {
-    const model = new StandardRouteWaypointModel(['BAKRR', 'S280']);
-    const spy = sinon.spy(model, '_setSpeedRestriction');
-
-    model._parseWaypointRestrictions('XYZ|S280');
-
-    t.true(spy.callCount === 1);
-    t.true(model._altitude === -1);
-    t.true(model._speed === 280);
+ava('._applyRestrictions() extracts all restrictions when non-ranged "AT/ABOVE" or "AT/BELOW" restrictions are used', (t) => {
+    const modelWithAltitude = new StandardRouteWaypointModel(['BAKRR', 'A100+']);
+    const modelWithSpeed = new StandardRouteWaypointModel(['BAKRR', 'S210-']);
+    const modelWithAltitudeAndSpeed = new StandardRouteWaypointModel(['BAKRR', 'A100-|S210+']);
```

Comment:
same here as with the altitude, I'd like to see this split into one test per case